### PR TITLE
feat(mcp): add core MCP server with docvet_check and docvet_rules tools

### LINKED ISSUE #260: Story 29.1: Core MCP Server Implementation
## Story

As an **AI agent (Claude Code, Cursor, etc.)**,
I want to invoke docvet checks via MCP protocol and receive structured findings,
so that I can analyze docstring quality programmatically without parsing CLI output.

## Acceptance Criteria

1. MCP client connects via stdio, calls `docvet_check` with path → runs all enabled checks, returns structured findings + summary
2. `docvet_check` with `checks` filter → only specified checks run
3. `docvet_check` with presence → includes `presence_coverage` object
4. `docvet_rules` tool → returns all 20 rules with name, check, description, category
5. Missing `mcp` dependency → clear ImportError with install instructions
6. Respects `[tool.docvet]` config (exclude patterns, thresholds)
7. Invalid requests → proper MCP error response (no crashes)

---

Story file: `_bmad-output/implementation-artifacts/29-1-core-mcp-server-implementation.md`
Branch: `feat/mcp-29-1-core-mcp-server-implementation`

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Adds a new `src/docvet/mcp.py` module that exposes docvet checks as MCP
tools over stdio. AI agents (Claude Code, Cursor, etc.) can now invoke
`docvet_check` to run docstring quality checks programmatically and
`docvet_rules` to retrieve the full 20-rule catalog — without parsing
CLI output.

- Add `mcp` optional dependency (`mcp>=1.6,<2`) to pyproject.toml
- Implement `docvet_check` tool: file/directory discovery, check dispatch (all except freshness by default), structured JSON response with findings, summary, and presence coverage
- Implement `docvet_rules` tool: static 20-rule catalog with name, check, description, category
- Add 45 unit tests covering all 7 ACs, error handling, schema parity, and edge cases
- Add `docvet.mcp` to module export tests in test_exports.py

Test: `uv run pytest tests/unit/test_mcp.py -v`

Closes #260

test(reporting): add boundary tests for exit code coverage threshold

fix(mcp): address code review findings for directory scoping and test gaps

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [x] Types pass (`uv run ty check`)
- [ ] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
- `src/docvet/mcp.py` — new module, follows LSP server pattern
- Directory scoping fix: `is_relative_to(target)` filter when path is a directory
- `_RULE_TO_CHECK` dict for O(1) rule-to-check mapping in summary aggregation
- Absolute paths in MCP findings are intentional (agents need unambiguous references)

### Related
- Story 29.2 will add CLI wiring (`docvet mcp` subcommand)
- Story 29.3 will add documentation and marketplace publishing

### CODE REVIEW

File: src/docvet/mcp.py [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+    _GRIFFE_AVAILABLE = True
+except ImportError:
+    _GRIFFE_AVAILABLE = False
+
+__all__ = ["start_server"]
+
+logger = logging.getLogger(__name__)
+
+# Default checks for MCP (freshness excluded — requires git context)
+_DEFAULT_MCP_CHECKS: frozenset[str] = _VALID_CHECK_NAMES - {"freshness"}
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/docvet/mcp.py]: Addressed in latest push — `_DEFAULT_MCP_CHECKS` now conditionally excludes griffe when `_GRIFFE_AVAILABLE` is False, so default calls don't produce spurious error entries.

File: src/docvet/mcp.py [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+    Examples:
+        Typically invoked by the ``docvet mcp`` CLI command:
+
+        ```python
+        from docvet.mcp import start_server
+
+        start_server()
+        ```
+    """
+    mcp_server.run(transport="stdio")
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/docvet/mcp.py]: Addressed in latest push — docstring now correctly states config is loaded per-request by each tool handler, not at server startup.
